### PR TITLE
Получение вакансий с сервера HH через API

### LINKED ISSUE #15: Получение вакансий с сервера HH через API

### PR DESCRIPTION
**Pull request by Squawk7777**
closes #15

### CODE REVIEW

File: src/main/java/com/jadajabajava/services/EmployerService.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+import java.util.Optional;
+
+@Slf4j
+@Service
+public class EmployerService extends AbstractService<Employer> {
+
+    private final EmployerRepository employerRepository;
+
+    private final MessageSource messageSource;
```
Comment (Squawk7777) [inline src/main/java/com/jadajabajava/services/EmployerService.java:18]: Codacy Issue found: [Perhaps 'messageSource' could be replaced by a local variable.](https://app.codacy.com/gh/jada-jaba-java/job-seeker-service/pullRequest?prid=5707872)

**Comment by Squawk7777** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- src/main/java/com/jadajabajava/services/EmployerService.java  1
- src/main/java/com/jadajabajava/entities/Skill.java  1
- src/main/java/com/jadajabajava/services/VacancyService.java  2
- src/main/java/com/jadajabajava/exceptions/VacancyProviderException.java  1
- src/main/java/com/jadajabajava/deserializers/HhVacancyDeserializer.java  3
- src/main/java/com/jadajabajava/entities/Salary.java  1
- src/main/java/com/jadajabajava/providers/HhVacancyProvider.java  4
- src/main/java/com/jadajabajava/exceptions/NotExistException.java  1
- src/main/java/com/jadajabajava/services/AbstractService.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/jada-jaba-java/job-seeker-service/pullRequest?prid=5707872&bid=18681037)

File: src/main/java/com/jadajabajava/services/EmployerService.java (hunk 1 above)
Comment (Squawk7777) [inline src/main/java/com/jadajabajava/services/EmployerService.java:18]: Codacy Issue found: [Avoid unused private fields such as 'messageSource'.](https://app.codacy.com/gh/jada-jaba-java/job-seeker-service/pullRequest?prid=5707872)

File: src/main/java/com/jadajabajava/JobSeekerService.java [hunk 2]
```diff
@@ -2,13 +2,14 @@
...
+import org.springframework.context.ConfigurableApplicationContext;
 import org.springframework.transaction.annotation.EnableTransactionManagement;
 
 @SpringBootApplication
 @EnableTransactionManagement
 public class JobSeekerService {
 
     public static void main(String[] args) {
-        SpringApplication.run(JobSeekerService.class, args);
+        ConfigurableApplicationContext ctx = SpringApplication.run(JobSeekerService.class, args);
```
Comment (Squawk7777) [inline src/main/java/com/jadajabajava/JobSeekerService.java:13]: Codacy Issue found: [Avoid unused local variables such as 'ctx'.](https://app.codacy.com/gh/jada-jaba-java/job-seeker-service/pullRequest?prid=5707872)

File: src/main/java/com/jadajabajava/providers/HhVacancyProvider.java [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+                .timeout(Duration.ofSeconds(TIMEOUT_SEC))
+                .build();
+
+        log.debug("Executing HTTP request: {}", uriComponents.toUri());
+        HttpResponse<String> httpResponse = httpClient.send(httpRequest, HttpResponse.BodyHandlers.ofString());
+        log.debug("Got response with code {} (body length: {})", httpResponse.statusCode(), httpResponse.body().length());
+        return mapper.readValue(httpResponse.body(), Vacancy.class);
+    }
+
+    private Set<Long> fetchVacancyIds(Integer startPageNumber, UriComponentsBuilder uriBuilder, HttpClient httpClient) throws IOException, InterruptedException {
```
Comment (Squawk7777) [inline src/main/java/com/jadajabajava/providers/HhVacancyProvider.java:113]: Codacy Issue found: [Avoid reassigning parameters such as 'startPageNumber'](https://app.codacy.com/gh/jada-jaba-java/job-seeker-service/pullRequest?prid=5707872)

**Comment by Squawk7777** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/com/jadajabajava/services/VacancyService.java  2
- src/main/java/com/jadajabajava/deserializers/HhVacancyDeserializer.java  3
- src/main/java/com/jadajabajava/entities/Skill.java  1
- src/main/java/com/jadajabajava/providers/HhVacancyProvider.java  5
- src/main/java/com/jadajabajava/services/AbstractService.java  1
- src/main/java/com/jadajabajava/services/EmployerService.java  1
- src/main/java/com/jadajabajava/entities/Salary.java  1
- src/main/java/com/jadajabajava/exceptions/VacancyProviderException.java  1
- src/main/java/com/jadajabajava/exceptions/NotExistException.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/jada-jaba-java/job-seeker-service/pullRequest?prid=5707872&bid=18681037)

**Comment by Squawk7777** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/com/jadajabajava/exceptions/VacancyProviderException.java  1
- src/main/java/com/jadajabajava/services/AbstractService.java  1
- src/main/java/com/jadajabajava/services/VacancyService.java  2
- src/main/java/com/jadajabajava/entities/Salary.java  1
- src/main/java/com/jadajabajava/services/EmployerService.java  1
- src/main/java/com/jadajabajava/entities/Skill.java  1
- src/main/java/com/jadajabajava/exceptions/NotExistException.java  1
- src/main/java/com/jadajabajava/providers/HhVacancyProvider.java  5
- src/main/java/com/jadajabajava/deserializers/HhVacancyDeserializer.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/jada-jaba-java/job-seeker-service/pullRequest?prid=5707872&bid=18681037)

**Comment by Squawk7777** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/com/jadajabajava/services/VacancyService.java  2
- src/main/java/com/jadajabajava/entities/Skill.java  1
- src/main/java/com/jadajabajava/exceptions/VacancyProviderException.java  1
- src/main/java/com/jadajabajava/services/AbstractService.java  1
- src/main/java/com/jadajabajava/deserializers/HhVacancyDeserializer.java  3
- src/main/java/com/jadajabajava/services/EmployerService.java  1
- src/main/java/com/jadajabajava/providers/HhVacancyProvider.java  5
- src/main/java/com/jadajabajava/exceptions/NotExistException.java  1
- src/main/java/com/jadajabajava/entities/Salary.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/jada-jaba-java/job-seeker-service/pullRequest?prid=5707872&bid=18681037)

**Review by bilarsen — COMMENTED** [review]
all good

**Pull request merged**